python/lsst/ts/observatory/control/utils/roi_spec.py

```diff
@@ -0,0 +1,79 @@
+# This file is part of ts_observatory_control.
+#
+# Developed for the Vera Rubin Observatory Telescope and Site.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = [
+    "ROICommon",
+    "ROI",
+    "ROISpec",
+]
+
+from typing import Any, Generator
+
+from pydantic import BaseModel, Field, validator
+
+
+class ROICommon(BaseModel):
+    rows: int = Field(default=50)
+    cols: int = Field(default=50)
+    integrationTimeMillis: int = Field(alias="integration_time_millis", default=100)
+
+    @validator("rows")
+    def check_rows(cls, v: int) -> int:
+        if v < 0:
```

Comment:
Is there any scenario where 0 is a valid minimum value? If not, maybe we should make these checks if v < 1.